src/frontend/youtube_clone/src/app/app.module.ts

```diff
@@ -1,16 +1,36 @@
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
-
+import { MatSlideToggleModule } from '@angular/material/slide-toggle';
 import { AppRoutingModule } from './app-routing.module';
+
 import { AppComponent } from './app.component';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MainPageComponent } from './main-page/main-page.component';
+import { VideoPageComponent } from './video-page/video-page.component';
+import { ProfilePageComponent } from './profile-page/profile-page.component';
+import { RouterModule } from '@angular/router';
+import {MatSidenavModule} from '@angular/material/sidenav';
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    MainPageComponent,
+    VideoPageComponent,
+    ProfilePageComponent,
   ],
   imports: [
     BrowserModule,
-    AppRoutingModule
+    AppRoutingModule,
+    BrowserAnimationsModule,
+    MatSlideToggleModule,
+    MatSidenavModule,
+    RouterModule.forRoot([      
+      {path: '', redirectTo: 'main-page', pathMatch: 'full'},
+      {path: 'main-page', component: MainPageComponent},
```

Comment:
If you visit YouTube and check their routes, they never mention `page` in the route name:

- https://www.youtube.com/
- https://www.youtube.com/feed/subscriptions
- https://www.youtube.com/feed/history

It doesn't make any sense because each route usually associated with a specific page

Reply:
Renamed route and components, so that it makes more sense and it's more similar to youtube.